secure/json-rpc/src/error.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use hex::FromHexError;
+use std::io;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum Error {
+    #[error("JSON RPC call {0} returned a custom internal error: {0}")]
+    InternalRPCError(String, String),
+    #[error("Data does not exist. Missing data: {0}")]
+    MissingData(String),
+    #[error("JSON RPC call {0} failed. Response: {0}")]
+    RPCFailure(String, String),
+    #[error("Serialization error: {0}")]
+    SerializationError(String),
+    #[error("Unknown error: {0}")]
+    UnknownError(String),
+}
+
+impl From<anyhow::Error> for Error {
```

Comment:
Please do not introduce anyhow into this crate unless it is blocked by a feature. I do not want anyhow ... if at all possible .. in our TCB.

Reply:
Motivations:
* Lots of unsafe code
* Implies soft definition of error types

Reply:
I was hoping to remove it entirely, but it is required for testing (e.g., to implement the DbReader for the mock database). It is also required if we want to handle anyhow errors automatically (e.g., using the '?' operator in Rust). However, I've managed to remove it from non-test dependencies by checking and handling each anyhow error manually now.